src/include/duckdb/main/appender.hpp

```diff
@@ -35,6 +35,9 @@ class Appender {
 	DUCKDB_API Appender(Connection &con, const string &table_name);
 	DUCKDB_API ~Appender();
 
+	//! The error string (in case execution was not successful)
+	string error;
```

Comment:
Since this is only required for the C API, could you instead create a wrapper for the Appender there and add the string error to that wrapper?